django-makemessages-xgettext/management/commands/xgettextmakemessages.py

```diff
@@ -0,0 +1,21 @@
+from optparse import make_option
+from django.core.management.commands import makemessages
+
+
+class Command(makemessages.Command):
+
+    option_list = makemessages.Command.option_list + (
+        make_option('--xgettext', '-X',
+                    default=[], dest='xgettext', action='append',
+                    help='Passes arguments to the xgettext command (e.g. '
+                         '--add-location=file). Use multiple times to pass '
+                         'additional arguments.'
+                    ),
+    )
+
+    def handle_noargs(self, *args, **options):
+
+        for option in options["xgettext"]:
+            self.xgettext_options = self.xgettext_options[:] + [option]
```

Comment:
If you take a look at makemessages.py:

    # Avoid messing with mutable class variables
    if options.get('no_wrap'):
        self.msgmerge_options = self.msgmerge_options[:] + ['--no-wrap']

I couldn't figure out why, but Django does it this way. I figured it was best to stay on track.

Reference: https://github.com/django/django/blob/stable/1.7.x/django/core/management/commands/makemessages.py#L221